src/box/lua/feedback_daemon.lua

```diff
@@ -357,7 +446,17 @@ local function guard_loop(self)
             self.fiber = fiber.create(feedback_loop, self)
             log.verbose("%s restarted", PREFIX)
         end
-        local st = pcall(fiber.sleep, self.interval)
+        if self.send_metrics and
+           get_fiber_id(self.metrics_collect_fiber) == 0 then
+            self.metrics_collect_fiber =
+                fiber.create(metrics_collect_loop, self)
+            log.verbose("%s restarted", METRICS_PREFIX)
+        end
+        local interval = self.interval
+        if self.send_metrics then
+            interval = math.min(interval, self.metrics_collect_interval)
```

Comment:
Not sure this is correct. Since the intervals are different, technically, I guess there should be two guard loops.

Reply:
I'm not even sure that we need this guard loop.

Reply:
I don't see its value in reconfiguration. I assume that it is needed to restart the fiber if it fails with an error. However, I don't see any calls that can cause an error.

Reply:
I've taken a close look at this function. And I still don't see any value in guard fiber.

Reply:
Sorry, my bad, I missed the fact `reload` also cancels the guard loop fiber.

But anyways, AFAIC, `json.encode` can throw an error inside the `feedback_loop` (e.g., on OOM) or `send_fiber` can get accidentally canceled by the user. The latter can also happen with the `metrics_collect_loop`.

Reply:
Leaving it up to @alyapunov.

Reply:
I think one guard is enough (and perhaps even one guard is more than needed).